### ISSUE TITLE
Add Microsite Class to Body

### ISSUE THREAD

**Issue by claytron** [collective/collective.lineage#4]
In Plone, classes are added to the body tag based on what section you are in or what template the current page uses (section-news, template-document_view). It would be great for a Lineage microsite to add a class as well, so that anywhere within a microsite, the class microsite-sitename is on the body tag.

**Comment by jensens** [comment]
I did this with the following code:

    from plone.app.layout.globals import layout as base
    class LayoutPolicy(base.LayoutPolicy):
        """Override the default Plone layout utility."""

        def bodyClass(self, template, view):
            """Include layout identifier in bodyClass."""
            body_class = super(LayoutPolicy, self).bodyClass(template, view)
            body_class += ' child-site'
            return body_class

and zcml::

    <browser:page
      allowed_interface="plone.app.layout.globals.interfaces.ILayoutPolicy"
      class=".layoutpolicy.LayoutPolicy"
      for="collective.lineage.interfaces.IChildSite"
      layer="kup.stadtentwicklungibk.interfaces.IKupStadtentwicklungibkLayer"
      name="plone_layout"
      permission="zope.Public"
    />

**Comment by thet** [comment]
@jensens - that would add it only to the IChildSite object, but not any other within the child site?
There might be the need to adapt it for any context and use ``current_childsite`` from the ``LineageUtils`` view in ``collective.lineage.browser``

**Comment by jensens** [comment]
Yes, the idea is to be able to detect the child-site itself, not if its in the child site.

In the child-sites the default ``site-NAME`` (which checks for ``INavigationRoot``) class is provided for the child site and all childs. 

We could think of providing two classes for maximum flexibilty: One ``child-site-root`` and one ``child-site``.

What I do not like is the overall approach to subclass and bind the view. I.e ``collective.cover`` provides also an own Layout adapter to add classes. We should think about a more dynamic hook in ``plone.app.layout`` to provide additional classes, for example simple named adapters which then be iterated in order to get the classes.
